maas-schemas/schemas/core/booking.json

```diff
@@ -69,15 +69,29 @@
       }
     },
     "cost": {
-      "$ref": "http://maasglobal.com/core/components/cost.json"
+      "oneOf": [
+        {
+          "$ref": "http://maasglobal.com/core/components/cost.json"
+        },
+        {
+          "type": "null"
+        }
+      ]
     },
     "configurator": {
       "$ref": "http://maasglobal.com/core/components/configurator.json"
     },
     "tspId": {
-      "type": "string",
-      "minLength": 1,
-      "maxLength": 256
+      "oneOf": [
```

Comment:
The `oneOf`s in `definitions` make it difficult to eliminate the `null` since there is no longer a way to talk about the value in situations where it can *not* be null. A proper fix would include adding separate cases for the null case, the not-null case and the maybe-null case. Moving the `oneOf` from definitions to `properties` is an easy compromise that doesn't corrupt the not-null type but also does not require you to invent new names for the null case and the combined case.